### PR TITLE
[mbr] Turn off interp inlining for hot reload

### PR DESCRIPTION
**Pull request by lambdageek**

### CODE REVIEW

**Comment by CoffeeFlux** [comment]
The only check for hot reload is the environment variable?

**Comment by lambdageek** [comment]
> The only check for hot reload is the environment variable?

Yes, for now.  This is an interim design.  The final design is in #47274 (but it's also converging on using an environment variable for at least some of the information that the runtime will receive)

**Comment by lambdageek** [comment]
by the way, for... reasons... the wasm code already has a separate check for this environment variable in c0f67b2191e166eac055b782d7e95c5feb6f2762

**Comment by naricc** [comment]
Why do we need to turn off inlining for hot reload? Just to keep things from secretly keeping a copy of things we want to reload? Or to prevent us needing to do a reload of everything that inlined a method we want to reload?

File: src/mono/mono/mini/mini-runtime.c [hunk 1]
```diff
@@ -4290,8 +4290,12 @@ mini_init (const char *filename, const char *runtime_version)
 
 	mono_interp_stub_init ();
 #ifndef DISABLE_INTERPRETER
-	if (mono_use_interpreter)
+	if (mono_use_interpreter) {
+		if (g_hasenv ("MONO_METADATA_UPDATE") && mono_interp_opts_string == NULL) {
+			mono_interp_opts_string = "-inline";
```
Comment (vargaz) [inline src/mono/mono/mini/mini-runtime.c:4295]: Isn't there a global variable which controls this instead of putting a getenv here ?

Comment (lambdageek) [reply src/mono/mono/mini/mini-runtime.c:4295]: No. the global variable is whether there's been any changes.  this is setting up startup.

**Comment by lambdageek** [comment]
> Why do we need to turn off inlining for hot reload? Just to keep things from secretly keeping a copy of things we want to reload? Or to prevent us needing to do a reload of everything that inlined a method we want to reload?

Yea it's so that we don't have to keep track of all the places where a given method has been inlined.  (ie: A calls B.  the interpreter inlines B into A.  I apply an update that changes B.  Now: (1) the runtime needs to know to invalidate the code for A; (2) the new code for B won't run until A finishes executing.

**Comment by lambdageek** [comment]
Going to close this.  I'll work on something that will support the new `DOTNET_MODIFIABLE_ASSEMBLIES`  environment var 
 (https://github.com/dotnet/runtime/issues/47274) and use it for all scenarios in a more componentized way.

**Pull request closed without merging**